test-lib.sh

```diff
@@ -311,6 +311,49 @@ EOF
   : "  Success!"
 }
 
+# ct_check_exec_env_vars
+# --------------------
+# Checks if all relevant environment variables from `docker run`
+# can be found in `docker exec` as well.
+# Uses: $CID_FILE_DIR - path to directory containing cid_files
+# Uses: $IMAGE_NAME - name of the image being tested
+function ct_check_exec_env_vars() {
+  local tmpdir exec_envs run_envs cid old_IFS
+  local var_name stripped filtered_envs
+  tmpdir=$(mktemp -d)
+  CID_FILE_DIR=${CID_FILE_DIR:-$(mktemp -d)}
+  # Get environment variables from `docker run`
+  docker run --rm "$IMAGE_NAME" /bin/bash -c "env" | sort > "$tmpdir/run_envs"
+  # Get environment variables from `docker exec`
+  ct_create_container "test_exec_envs" bash -c "sleep 1000" >/dev/null
```

Comment:
Based on the PR https://github.com/sclorg/s2i-nodejs-container/pull/230 container is already running by this command https://github.com/sclorg/s2i-nodejs-container/blob/f4beccc99db5dde479965dccebe3f655cac838c9/test/run#L440 so you can only connect to already exists container and not create the new one. cid_file is already present and you can passed it as argument https://github.com/sclorg/s2i-nodejs-container/blob/f4beccc99db5dde479965dccebe3f655cac838c9/test/run#L448

Reply:
That is a possibility, but I would rather have the test case standalone for cases where no such running container is available. Considering we do not really wait on the container finishing some work I do not think it much of a problem to create a new one every time.